implementation/src/main/java/io/smallrye/faulttolerance/config/SecurityActions.java

```diff
@@ -98,10 +100,20 @@ private static Method doGetMethod(final Class<?> beanClass, final Class<?> decla
         ParametrizedParamsTranslator expectedParamsTranslator = prepareParamsTranslator(beanClass, declaringClass);
 
         ParametrizedParamsTranslator actualParamsTranslator = new ParametrizedParamsTranslator();
+
+        // gather all interface methods based on declaring class
+        Set<String> allInterfaceMethodNames = new HashSet<>();
+        for (Class<?> clazz : declaringClass.getInterfaces()) {
```

Comment:
Should this perhaps include super-interfaces of `declaringClass.getInterfaces()`?

Reply:
I don't think spec allows that. Quoting:

> The named fallback method must be on the same class, a superclass or an implemented interface of the class which declares the annotated method

From `implemented interface of the class...` I assume it has to be on an interface that directly implemented by given class.

Reply:
I guess that's probably yet another vague formulation in the spec, but fair enough.

Reply:
Could be the case, to be fair I am learning about FT spec "on the fly" so take what I say with a pinch of salt ;)